signer/signer.go

```diff
@@ -36,24 +36,34 @@ import (
 // signingAgent is the unprotected header field used by signature.
 const signingAgent = "Notation/1.0.0"
 
-// genericSigner implements notation.Signer and embeds signature.Signer
-type genericSigner struct {
+// GenericSigner implements notation.Signer and embeds signature.Signer
+type GenericSigner struct {
 	signature.Signer
 }
```

Comment:
Since `GenericSigner` is exposed, do you want to hide `signature.Signer` from public use?